jetty-websocket/websocket-core-server/src/main/java/org/eclipse/jetty/websocket/core/server/ServerUpgradeRequest.java

```diff
@@ -90,9 +91,17 @@ public ServerUpgradeRequest(WebSocketNegotiation negotiation) throws BadMessageE
     {
         if (cookies == null)
         {
-            cookies = Arrays.stream(request.getCookies())
-                .map(c -> new HttpCookie(c.getName(), c.getValue()))
-                .collect(Collectors.toList());
+            Cookie[] reqCookies = request.getCookies();
+            if (reqCookies != null)
+            {
+                cookies = Arrays.stream(reqCookies)
+                    .map(c -> new HttpCookie(c.getName(), c.getValue()))
+                    .collect(Collectors.toList());
+            }
+            else
+            {
+                cookies = List.of();
```

Comment:
Shouldn't we also return null here if the request had no cookies?

Reply:
We had this discussion already recently and determined that `emptyList` was both more readable and more effficient.

```suggestion
                cookies = Collections.emptyList();
```